Library/OcDeviceMiscLib/SetResizableBar.c

```diff
@@ -388,3 +689,178 @@ ResizeGpuBars (
+  UINTN                              PciAddress;
+
+  ASSERT (Size < PciBarTotal);
+
+  Status = gBS->LocateHandleBuffer (
+                  ByProtocol,
+                  &gEfiPciRootBridgeIoProtocolGuid,
+                  NULL,
+                  &HandleCount,
+                  &HandleBuffer
+                  );
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "OCDM: No PCI devices for RBAR support - %r\n", Status));
+    return Status;
+  }
+
+  for (Index = 0; Index < HandleCount; ++Index) {
+    Status = gBS->HandleProtocol (
+                    HandleBuffer[Index],
+                    &gEfiPciRootBridgeIoProtocolGuid,
+                    (VOID **)&PciRootBridgeIo
+                    );
+    if (EFI_ERROR (Status)) {
+      continue;
+    }
+
+    PciRootBridgeIo->Configuration (PciRootBridgeIo, (VOID **)&Descriptors);
+    if (EFI_ERROR (Status)) {
+      continue;
+    }
+
+    // Not sure if multiple root bridge systems even exist but this should support them
+    while (TRUE) {
+      Status = PciGetNextBusRange (&Descriptors, &MinBus, &MaxBus, &IsEnd);
+      if (EFI_ERROR (Status)) {
+        break;
+      }
+
+      if (IsEnd || (Descriptors == NULL)) {
+        break;
+      }
+
+      for (Bus = 0; Bus <= MaxBus; Bus++) {
+        for (Dev = 0; Dev <= PCI_MAX_DEVICE; Dev++) {
+          for (Func = 0; Func <= PCI_MAX_FUNC; Func++) {
+            PciAddress = EFI_PCI_ADDRESS (Bus, Dev, Func, 0);
+
+            // PciAddrOffset doesnt need to be used below 256
+            Status = PciRootBridgeIo->Pci.Read (
+                                            PciRootBridgeIo,
+                                            EfiPciWidthUint8,
+                                            PciAddress + PCI_CLASSCODE_OFFSET,
+                                            sizeof (PCI_CLASSCODE) / sizeof (UINT8),
+                                            &ClassCode
+                                            );
+            if (EFI_ERROR (Status)) {
+              continue;
+            }
+
+            DEBUG ((DEBUG_VERBOSE, "OCDM: PCI device %u/%u/%u has class %X\n", Bus, Dev, Func, ClassCode));
+
+            if (ClassCode.BaseCode != PCI_CLASS_DISPLAY) {
+              continue;
+            }
+
+            DEBUG ((
+              DEBUG_INFO,
+              "OCDM: Setting RBAR to %u inc %d on %u/%u/%u\n",
+              Size,
+              Increase,
+              Bus,
+              Dev,
+              Func
+              ));
+            Status = SetResizableBarOnDeviceRbIo (PciRootBridgeIo, PciAddress, Size, Increase);
+            if (!EFI_ERROR (Status)) {
+              HasSuccess = TRUE;
+            }
+
+            PciRootBridgeIo->Pci.Read (PciRootBridgeIo, EfiPciWidthUint8, PciAddress + PCI_HEADER_TYPE_OFFSET, 1, &HdrType);
+            if (!Func && ((HdrType & HEADER_TYPE_MULTI_FUNCTION) == 0)) {
+              break;
+            }
+          }
+        }
+      }
+    }
+  }
+
```

Comment:
HandleBuffer is leaked

Reply:
This issue also exists in the PciIo version. I'm unable to test that on my system but it would need to be fixed too